syntax = "proto3";

package filetransfer;

service FileTransfer {
  rpc Upload(stream FileChunk) returns (TransferStatus);
  rpc Download(FileRequest) returns (stream FileChunk);
}

message FileChunk {
  string filename = 1;
  bytes content = 2;
  bool is_dir = 3;
  uint64 filesize = 4;
}

message FileRequest {
  string path = 1;
}

message TransferStatus {
  bool success = 1;
  string message = 2;
}
